src/storage.rs

```diff
@@ -13,6 +14,29 @@ use super::errors::Error;
 pub type DatabaseConnection = r2d2::PooledConnection<SqliteConnectionManager>;
 pub type DatabaseConnectionPool = r2d2::Pool<SqliteConnectionManager>;
 
+#[derive(PartialEq, Eq, Hash)]
+pub struct RoomId {
+    id: String,
+}
+
+impl RoomId {
+    pub fn new(room_id: &str) -> RoomId {
+        let whitespace = Regex::new(r"/\s+/g").expect("regex is valid");
+        let without_whitespace = whitespace.replace_all(&room_id, "-").to_string();
+        let special_characters = Regex::new(r"/[^\w-]/g").expect("regex is valid");
+        let parsed_id = special_characters.replace_all(&without_whitespace, "-").to_string();
+        RoomId { id: parsed_id }
+    }
+
+    pub fn get_id(&self) -> &str {
+        &self.id
+    }
+
+    pub fn to_string(&self) -> String {
```

Comment:
You probably don't need `to_string` because you can always construct one from `&str` returned from `get_id`